### ISSUE TITLE
Docker Desktop for windows makes a mess of my hosts file.

### ISSUE THREAD

**Issue by resc** [docker/for-win#5855]
- [* ] I have tried with the latest version of my channel (Stable or Edge)
  - [ ] I have uploaded Diagnostics
  - Diagnostics ID:

### Expected behavior
My c:\windows\system32\drivers\etc\hosts file only contains entries for the current set of ip-adresses of my machine for host.docker.internal and gateway.docker.internal

I expect my hosts file to look like this:
```
# Copyright (c) 1993-2009 Microsoft Corp.
#
# This is a sample HOSTS file used by Microsoft TCP/IP for Windows.
#
# This file contains the mappings of IP addresses to host names. Each
# entry should be kept on an individual line. The IP address should
# be placed in the first column followed by the corresponding host name.
# The IP address and the host name should be separated by at least one
# space.
#
# Additionally, comments (such as these) may be inserted on individual
# lines or following the machine name denoted by a '#' symbol.
#
# For example:
#
#      102.54.94.97     rhino.acme.com          # source server
#       38.25.63.10     x.acme.com              # x client host

# localhost name resolution is handled within DNS itself.
#	127.0.0.1       localhost
#	::1             localhost
# Added by Docker Desktop
192.168.21.129 host.docker.internal
192.168.21.129 gateway.docker.internal
# To allow the same kube context to work on the host and the container:
127.0.0.1 kubernetes.docker.internal
# End of section
```
### Actual behavior
my hosts file looks like this:
Note that somewhere in the middle the hostnames appears without ip-addresses too, that's not a copy-paste error.
```
# Copyright (c) 1993-2009 Microsoft Corp.
#
# This is a sample HOSTS file used by Microsoft TCP/IP for Windows.
#
# This file contains the mappings of IP addresses to host names. Each
# entry should be kept on an individual line. The IP address should
# be placed in the first column followed by the corresponding host name.
# The IP address and the host name should be separated by at least one
# space.
#
# Additionally, comments (such as these) may be inserted on individual
# lines or following the machine name denoted by a '#' symbol.
#
# For example:
#
#      102.54.94.97     rhino.acme.com          # source server
#       38.25.63.10     x.acme.com              # x client host

# localhost name resolution is handled within DNS itself.
#	127.0.0.1       localhost
#	::1             localhost
10.0.6.214 host.docker.internal
10.0.6.214 gateway.docker.internal
10.0.7.192 host.docker.internal
10.0.7.192 gateway.docker.internal
192.168.18.227 host.docker.internal
192.168.18.227 gateway.docker.internal
192.168.21.77 host.docker.internal
192.168.21.77 gateway.docker.internal
192.168.21.84 host.docker.internal
192.168.21.84 gateway.docker.internal
 host.docker.internal
 gateway.docker.internal
192.168.0.156 host.docker.internal
192.168.0.156 gateway.docker.internal
192.168.0.194 host.docker.internal
192.168.0.194 gateway.docker.internal
192.168.21.128 host.docker.internal
192.168.21.128 gateway.docker.internal
192.168.21.101 host.docker.internal
192.168.21.101 gateway.docker.internal
# Added by Docker Desktop
192.168.21.129 host.docker.internal
192.168.21.129 gateway.docker.internal
# To allow the same kube context to work on the host and the container:
127.0.0.1 kubernetes.docker.internal
# End of section
```

### Information

  - Windows Version: Win 10 Pro Version 2004 build 19041.84
  - Docker Desktop Version: 2.2.0.3 (42716) stable channel
  - Are you running inside a virtualized Windows e.g. on a cloud server or on a mac VM: No

### Steps to reproduce the behavior

  1. install Docker Desktop for windows
  2. Stop Docker Desktop for windows while laptop is connected one physical network
  3. Start Docker Desktop for windows while laptop is connected another physical network
  4. Repeat on more networks to get more entries in hosts

**Comment by stephen-turner** [comment]
Thanks for the report. This is fixed in our latest development copy already. It will be in the next Edge release.

**Comment by MikeWilliams-UK** [comment]
Has this issue been fixed?
I am running 2.2.0.5 (43884) and my hosts file has been messed up.

Not sure when this happened, am I correct that in my reading that the entries inserted/maintained by docker are bracketed by the following lines
"# Added by Docker Desktop"
"# End of section"

**Comment by stephen-turner** [comment]
Thanks for pinging this. I believe it was fixed in Edge 2.2.1.0, but we forgot to add it to the release notes so the ticket didn't get closed. I'll close it now. I think it's not yet on Stable, but will be in the next Stable.
